src/libraries/Common/src/System/Security/Cryptography/ECDiffieHellmanOpenSsl.cs

```diff
@@ -95,7 +95,9 @@ public override ECDiffieHellmanPublicKey PublicKey
             get
             {
                 ThrowIfDisposed();
-                return new ECDiffieHellmanOpenSslPublicKey(_key.UpRefKeyHandle());
+
+                using SafeEvpPKeyHandle handle = _key.UpRefKeyHandle();
```

Comment:
This one seems interesting.

The constructor doesn't up-ref the key:

https://github.com/dotnet/runtime/blob/8e4fe9a21ba888559512dd8d9feae39658edfbe8/src/libraries/Common/src/System/Security/Cryptography/ECDiffieHellmanOpenSslPublicKey.cs#L13-L21

And nor does ECOpenSsl:

https://github.com/dotnet/runtime/blob/8e4fe9a21ba888559512dd8d9feae39658edfbe8/src/libraries/Common/src/System/Security/Cryptography/ECOpenSsl.cs#L28-L30

Where or how is ECDiffieHellmanOpenSslPublicKey taking owenership of the handle passed to it?

Reply:
https://github.com/dotnet/runtime/blob/a5f3676cc71e176084f0f7f1f6beeecd86fbeafc/src/libraries/Common/src/System/Security/Cryptography/ECOpenSsl.cs#L54-L57
?

Reply:
OK, I think I see the flow.  Assume we're at a refcount of 1 when we start.

* get_PublicKey
  * _key.UpRefKeyHandle
    * EVP_PKEY_set1_EC_KEY => refcount 2
  * ECDiffieHellmanOpenSslPublicKey..ctor
    * EVP_PKEY_get1_EC_KEY => refcount 3
    * instance holds the key
  * Steve's new deterministic call to SaveEvpPKeyHandle.Dispose()
    * EVP_PKEY_free => the EVP_PKEY refcount goes to 0
      * EVP_PKEY_free_it => refcount 2
* Done.  Final refcount: 2.  Expected refcount: 2.  Happiness.

Reply:
Yeah, the comment above `EvpPkeyGetEcKey` lead me astray a bit. Now that I looked at how `EvpPkeyGetEcKey` works, it's `EVP_PKEY_get1_EC_KEY`. So the "1" means it returns an owned handle.